src/Symfony/Component/Form/Extension/Core/DataTransformer/ValueToSignedTransformer.php

```diff
@@ -0,0 +1,135 @@
+    }
+
+    /**
+     * Sign the given value and return an array with the value and the signature.
+     *
+     * @param mixed $value The value
+     *
+     * @return array The array
+     */
+    public function transform($value)
+    {
+        return array(
+            $this->fieldKey => $value,
+            $this->signedKey => $this->calculateSignature($value),
+        );
+    }
+
+    /**
+     * Extracts the value and check signature from an array.
+     *
+     * @param array $array
+     *
+     * @return mixed The value
+     *
+     * @throws TransformationFailedException If the given value is not an array or
+     *                                       if the given array can not be transformed.
+     */
+    public function reverseTransform($array)
+    {
+        if (!is_array($array)) {
+            throw new TransformationFailedException('Expected an array.');
+        }
+
+        if (!array_key_exists($this->fieldKey, $array) || !array_key_exists($this->signedKey, $array)) {
+            throw new TransformationFailedException('Expected an array with data and signature.');
+        }
+
+        $newSignature = $this->calculateSignature($array[$this->fieldKey]);
+
+        if (!$this->checkSignature($array[$this->signedKey], $newSignature)) {
+            throw new TransformationFailedException(
+                'The signature does not match with the provided data.'
+            );
+        }
+
+        return $array[$this->fieldKey];
+    }
+
+    /**
+     * Calculate signature for the provided data based on the configured algo & secret.
+     *
+     * @return string the calculated signature
+     */
+    protected function calculateSignature($value)
+    {
+        $ctx = hash_init($this->algo, HASH_HMAC, $this->secret);
+
+        if (!is_array($value)) {
+            $value = array($value);
+        }
+
+        foreach ($value as $valueElement) {
+            hash_update($ctx, $valueElement);
+        }
+
+        return hash_final($ctx);
+    }
+
+    /**
+     * Check two signature with protection on timing attack.
+     *
+     * @param $signatureA
+     * @param $signatureB
+     *
+     * @return bool
+     */
+    protected static function checkSignature($signatureA, $signatureB)
+    {
+        if (!is_string($signatureA) || !is_string($signatureB)) {
+            return false;
+        }
+
+        /*
+         * Trying to avoid timing attack due to the behavior of string equality condition (==)
+         *
+         * In certain cases, information can be leaked by using a timing attack.
+         * It takes advantage of the == operator only comparing until it finds a difference in the two strings. To prevent it, you have two options.
+         */
+        if (($len = strlen($signatureA)) != strlen($signatureB)) {
+            return false;
+        }
+
+        $status = 0;
+        for ($i = 0; $i < $len; ++$i) {
+            $status |= ord($signatureA[$i]) ^ ord($signatureB[$i]);
```

Comment:
you're right, i've missed that we have the polyfill form php56 function like hash_equals.